packages/zosjobs/src/ModifyJobs.ts

```diff
@@ -0,0 +1,118 @@
+     * @memberof ModifyJobs
+     */
+    public static async modifyJob(session: AbstractSession, jobname: string, jobid: string,
+        jobclass?: string, hold?: boolean, release?: boolean): Promise<undefined|IJobFeedback> {
+        this.log.trace("ModifyJob called with jobname %s jobid %s", jobname, jobid);
+        return ModifyJobs.modifyJobCommon(session, { jobname, jobid, jobclass, hold, release});
+    }
+
+    /**
+     * Modify a job
+     * @static
+     * @param {AbstractSession} session - z/OSMF connection info
+     * @param {IJob} job - the job that you want to modify
+     * @returns {Promise<undefined|IJobFeedback>} - promise of undefined, or IJobFeedback object returned by API if modifyVersion is 2.0
+     * @memberof ModifyJobs
+     */
+    public static async modifyJobForJob(session: AbstractSession, job: IJob ): Promise<undefined|IJobFeedback> {
+        this.log.trace("ModifyJobForJob called with job %s", JSON.stringify(job));
+        return ModifyJobs.modifyJobCommon(session,
+            { jobname:job.jobname, jobid: job.jobid, jobclass: job.class, hold: job.hold, release: job.release });
+    }
+
+    /**
+     * Modify a job
+     * Full version of the API with a parameter object
+     * @static
+     * @param {AbstractSession} session - z/OSMF connection info
+     * @param {IModifyJobParms} parms - parm object (see IModifyJobParms interface for details)
+     * @returns {Promise<undefined|IJobFeedback>} - promise of undefined, or IJobFeedback object returned by API if modifyVersion is 2.0
+     * @memberof ModifyJobs
+     */
+    public static async modifyJobCommon(session: AbstractSession, parms: IModifyJobParms): Promise<undefined|IJobFeedback> {
+        this.log.trace("ModifyJobCommon called with parms %s", JSON.stringify(parms));
+        ImperativeExpect.keysToBeDefinedAndNonBlank(parms, ["jobid", "jobname"],
+            "You must specify both the jobname and jobid for the job you want to modify.");
+        this.log.info("Modifying job %s.%s", parms.jobname, parms.jobid);
+
+        const headers: any = [Headers.APPLICATION_JSON];
+        const parameters: string = "/" + parms.jobname + "/" + parms.jobid;
+        let response: IJobFeedback;
+        let request: IModifyJob;
+        let mergedMessage: string = "";
+        let exception: boolean = false;
+
+        if(parms.hold && parms.release){
+            throw new ImperativeError({msg: "Parameters `hold` and `release` are in conflict and cannot be specified together"});
+        }
+
+        if(parms.hold || parms.release){
+            parms.hold ? request = { request: "hold"} : request = { request: "release"};
+            try{
+                response = await ZosmfRestClient.putExpectJSON(session, JobsConstants.RESOURCE + parameters, headers, request);
+            }
+            catch(err){
+                exception = true;
+                err.mMessage=err.mMessage.concat('Modification Error');
+                return err;
```

Comment:
Please consider throwing the error 😋 

---

`When` using the API `ModifyJobs.modifyJob(...)`
`If   `&nbsp;&nbsp; the first request (hold/release) throws an error (e.g ECONNREFUSED)
`Then` the type of the resolved value is not `IJobFeedback` anymore.

This results in an error like the one bellow
```
Unexpected Command Error:
Please review the message and stack below.
Contact the creator of handler:
"/root/gh/zowe/zowe-cli/packages/cli/lib/zosjobs/modify/job/Job.handler"
Message:
Cannot set property message of Error which has only a getter
Stack:
TypeError: Cannot set property message of Error which has only a getter
    at JobHandler.<anonymous> (/root/gh/zowe/zowe-cli/packages/cli/lib/zosjobs/modify/job/Job.handler.js:67:34)
    at Generator.next (<anonymous>)
    at fulfilled (/root/gh/zowe/zowe-cli/packages/cli/lib/zosjobs/modify/job/Job.handler.js:15:58)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```